state/tx_config.go

```diff
@@ -0,0 +1,187 @@
+package state
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+
+	"github.com/cosmos/cosmos-sdk/crypto/keyring"
+	sdktypes "github.com/cosmos/cosmos-sdk/types"
+
+	"github.com/celestiaorg/celestia-app/pkg/user"
+	apptypes "github.com/celestiaorg/celestia-app/x/blob/types"
+)
+
+const (
+	// DefaultGasPrice specifies the default gas price value to be used when the user
+	// wants to use the global minimal gas price, which is fetched from the celestia-app.
+	DefaultGasPrice float64 = -1.0
+	// gasMultiplier is used to increase gas limit in case if tx has additional cfg.
+	gasMultiplier = 1.1
+)
+
+// NewTxConfig constructs a new TxConfig with the provided options.
+// It starts with a DefaultGasPrice and then applies any additional
+// options provided through the variadic parameter.
+func NewTxConfig(opts ...ConfigOption) *TxConfig {
+	options := &TxConfig{gasPrice: DefaultGasPrice}
+	for _, opt := range opts {
+		opt(options)
+	}
+	return options
+}
+
+// TxConfig specifies additional options that will be applied to the Tx.
+type TxConfig struct {
+	// Specifies the address from the keystore that will sign transactions.
+	// NOTE: Only `signerAddress` or `KeyName` should be passed.
+	// signerAddress is a primary cfg. This means If both the address and the key are specified,
+	// the address field will take priority.
+	signerAddress string
+	// Specifies the key from the keystore associated with an account that
+	// will be used to sign transactions.
+	// NOTE: This `Account` must be available in the `Keystore`.
+	keyName string
+	// gasPrice represents the amount to be paid per gas unit.
+	// Negative gasPrice means user want us to use the minGasPrice
+	// defined in the node.
```

Comment:
nit: minGasPrice is a bit ambiguous because in celestia-app version >= 2, consensus nodes have a local min gas price. There is also a global network min gas price which is a governance modifiable parameter so perhaps it is worth clarifying which gas price this defaults to: the local min gas price or the network min gas price.